File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    vector<int> memo = vector<int>(46, -1);
    int climbStairs(int n) {
        if (n < 0) return 0;
        if (n == 0) return 1;
        if (memo[n] != -1) return memo[n];
        
        int one = climbStairs(n - 1);
        int two = climbStairs(n - 2);
        return memo[n] = one + two;
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    int [] dp = new int[46];
    int solve(int n){
        if(n == 0) return 1;
        if(n < 0) return 0;
        if(dp[n] != -1) return dp[n];
        int op1 = solve(n-1);
        int op2 = solve(n-2);
        return dp[n] = op1+op2;
    }
    public int climbStairs(int n) {
        Arrays.fill(dp, -1);
        return solve(n);
    }
}